### ISSUE TITLE
'Special' keymap-prefixes don't work

### ISSUE THREAD

**Issue by maxi-k** [hrs/engine-mode#17]
When changing the keymap-prefix, I found out that some keys don't work as described in the README. For example:
```elisp
;; Doesn't work
(setq engine/keymap-prefix (kbd "s-/"))
;; Error
;; Wrong type argument: characterp, 8388655
```
(the 's-' is the ```cmd``` key on macs)

Just using a bare string doesn't give an error and the keybinding works:
```elisp
;; Works
(setq engine/keymap-prefix "s-/")
```

The issue seems to be the ```engine/scope-keybinding``` function, because it concats the prefix and the engine keybinding, but the ```kbd``` macro doesn't always return a string.

It's a minor issue that can be easily worked around, just wanted to inform you.

**Comment by hrs** [comment]
Hey, @maxi-k! Thanks for the letting me know about that—it's definitely a bug. I just pushed PR #20, which swaps out that sloppy string concatenation logic for real keymaps.

There's a tradeoff in that PR, which is that `engine/keymap-prefix` will be non-backwards-compatibly replaced with `engine/set-keymap-prefix`, since it'll need to bind a keymap. However, both of these will then work:
```emacs
(engine/set-keymap-prefix "s-/")
(engine/set-keymap-prefix (kbd "s-/"))
```
Anyway, feel free to take a look at the PR and let me know if it addresses your concerns. Otherwise I'll probably merge it into master after a day or two. Thanks again!

**Comment by maxi-k** [comment]
Thanks for the quick answer & fix!

It does work now when using the ```kbd``` macro.

Without ```kbd```, it binds the prefix to the combination ```s - /```, as in pressing ```s minus slash``` successively (this is the case for other bindings like ```C-s``` too).
It *should* throw an error saying ```"Key sequence s - / starts with non-prefix key s"```, but it doesn't in this case. I have no idea why.

This seems to be an emacs bug though, or something is wrong with my config (which is much more probable).

As for backwards compatibility, you could maybe do something like this, if it's a big concern:
```elisp
(add-hook 'after-init-hook
          (lambda ()
            (when (boundp 'engine/keymap-prefix)
              (message "Engine mode: You are using an obsolete way of binding the prefix key. Please use engine/set-keymap-prefix instead")
              (engine/set-keymap-prefix engine/keymap-prefix))))
```
Just a thought, it's probably not very practical.

Thanks so much again!

**Comment by hrs** [comment]
Cool, I'm glad to hear that the fix works with the `kbd` macro! It's unfortunate that it's not working correctly without it, though. I'm not sure how I missed that when I was first testing it, but after trying it again you're totally right.

That `after-init-hook` seems reasonable, but I think I'm going to leave it out... I'm not sure that it's a big enough problem to warrant adding a shim. If I get any complaints, though, I'll probably add it in.

**Comment by hrs** [comment]
Fixed (mostly) in #20.
